util/util.go

```diff
@@ -294,9 +310,9 @@ func referSameObject(a, b metav1.OwnerReference) bool {
 }
 
 // PointsTo returns true if any of the owner references point to the given target
-func PointsTo(refs []metav1.OwnerReference, target *metav1.ObjectMeta) bool {
+func PointsTo(refs []metav1.OwnerReference, target metav1.Object) bool {
```

Comment:
PointsTo should likely use similar logic to `referSameVersion` above, since UID cannot be relied upon when taking pivoting into account.